expression/expression.go

```diff
@@ -111,26 +111,34 @@ func (e CNFExprs) Clone() CNFExprs {
 	return cnf
 }
 
-// EvalBool evaluates expression list to a boolean value.
-func EvalBool(ctx sessionctx.Context, exprList CNFExprs, row chunk.Row) (bool, error) {
+// EvalBool evaluates expression list to a boolean value. The first returned value
+// indicates bool result of the expression list, the second returned value indicates
+// whether the result of the expression list is null, it can only be true when the
+// first returned values is false.
+func EvalBool(ctx sessionctx.Context, exprList CNFExprs, row chunk.Row) (bool, bool, error) {
+	hasNull := false
 	for _, expr := range exprList {
 		data, err := expr.Eval(row)
 		if err != nil {
-			return false, err
+			return false, false, err
 		}
 		if data.IsNull() {
-			return false, nil
+			hasNull = true
```

Comment:
If the first expr is `null`, while the second expr is `false`, we should return `selected = false, isNull = false, error = nil`, because `null and false` is `false`, not `null`.